src/sinks/socket.rs

```diff
@@ -164,63 +159,97 @@ mod test {
     // This is a test that checks that we properly receieve all events in the
     // case of a proper server side write side shutdown.
     //
-    // This test basically sends 10 events shutsdown the server and forces a
+    // This test basically sends 10 events, shutsdown the server and forces a
     // reconnect. It then forces another 10 events through and we should get a
     // total of 20 events.
     //
     // If this test hangs that means somewhere we are not collecting the correct
     // events.
+    #[cfg(feature = "sources-tls")]
     #[test]
     fn tcp_stream_detects_disconnect() {
+        use crate::tls::{MaybeTlsSettings, TlsConfig, TlsOptions};
```

Comment:
whats the reason for bringing these imports down here? Is it for the feature flag?

Reply:
Yes, it's under `sources-tls` feature, otherwise `check-component-features` will fail. If I move everything what not under this feature, then not used errors will be thrown.